pkg/registry/apis/provisioning/resources/dualwriter.go

```diff
@@ -162,34 +156,18 @@ func (r *DualReadWriter) CreateResource(ctx context.Context, path string, ref st
 		Ref:  ref,
 	}
 
-	// TODO: improve parser to parse out of reader
-	parsed, err := r.parser.Parse(ctx, info, true)
+	parsed, err := r.parser.Parse(ctx, info)
 	if err != nil {
-		if errors.Is(err, ErrUnableToReadResourceBytes) {
-			return nil, apierrors.NewBadRequest("unable to read the request as a resource")
-		}
-
-		return nil, err
-	}
-
-	// GVR will exist for anything we can actually save
-	// TODO: Add known error in parser for unsupported resource
-	if parsed.GVR == nil {
-		return nil, apierrors.NewBadRequest("The payload does not map to a known resource")
-	}
-
-	// Do not write if any errors exist
-	if len(parsed.Errors) > 0 {
-		return parsed, err
+		return nil, fmt.Errorf("parse file: %w", err)
 	}
 
 	data, err = parsed.ToSaveBytes()
 	if err != nil {
 		return nil, err
 	}
 
-	err = r.repo.Create(ctx, path, ref, data, message)
-	if err != nil {
+	// TODO: do we want to do a dry run before saving into the repository?
```

Comment:
I don't see how that would help....  the DryRun is useful to know if we *could* save without actually saving.  When actually saving we can just return the errors directly